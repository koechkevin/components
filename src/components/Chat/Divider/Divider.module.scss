@import '../../../styles/global';

.divider {
  padding: 0 32px;
  font-size: 14px;
  font-weight: 700;

  &:global(.ant-divider-horizontal.ant-divider-with-text-center) {
    margin: 8px 0;
    font-weight: bold;
    font-size: 14px;
  }

  @include respond-to('sm') {
    padding: 0;
  }
}
