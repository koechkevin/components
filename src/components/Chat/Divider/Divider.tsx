import { Divider as AntDivider } from 'antd';
import React, { CSSProperties, FC } from 'react';

import { dateFormat } from '../../../utils';

import styles from './Divider.module.scss';

interface Props {
  style?: CSSProperties;
  dateTime: Date;
}

const Divider: FC<Props> = (props) => {
  const { style } = props;

  return (
    <AntDivider className={styles.divider} style={style}>
      {dateFormat(props.dateTime)}
    </AntDivider>
  );
};

export default Divider;
