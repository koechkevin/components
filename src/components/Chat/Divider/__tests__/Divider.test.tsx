import { Divider as AntdDivider } from 'antd';
import { shallow } from 'enzyme';
import React from 'react';

import Divider from '../Divider';

describe('Tests for chat divider', () => {
  const props = {
    dateTime: new Date(),
  };

  const wrapper = shallow(<Divider {...props} />);

  it('should test for mounting of divider component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render antd divider', () => {
    expect(wrapper.find(AntdDivider).exists()).toEqual(true);
  });

  it('should render dateFormat time in divider', () => {
    expect(wrapper.contains('Today')).toEqual(true);
  });

  // FIXME: fix this test case
  // it('should render Just now text', () => {
  //   expect(wrapper.contains('Just now')).toEqual(true);
  // });
});
