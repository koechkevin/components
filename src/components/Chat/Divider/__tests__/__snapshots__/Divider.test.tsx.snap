// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for chat divider should test for mounting of divider component 1`] = `ShallowWrapper {}`;
